src/main/java/teammates/common/datatransfer/CommentAttributes.java

```diff
@@ -48,7 +48,7 @@ public CommentAttributes(String courseId, String giverEmail, CommentParticipantT
         this.giverEmail = giverEmail;
         this.recipientType = recipientType == null ? CommentParticipantType.PERSON : recipientType;
         this.recipients = recipients;
-        this.commentText = commentText;
+        this.commentText = commentText == null ? null : new Text(Sanitizer.sanitizeForRichText(commentText.getValue()));
```

Comment:
@damithc , we have discussed to do it in a separate issue in a PR which was accidentally closed. This one is huge and it is getting more difficult to rebase it, so let's merge this part